k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -284,9 +284,19 @@ void EDM4hep2LcioTool::convertAdd(const std::string& e4h_coll_name, const std::s
                                   std::vector<EDM4hep2LCIOConv::ParticleIDConvData>& pidCollections) {
   const auto& evtFrame = m_podioDataSvc->getEventFrame();
   const auto& metadata = m_podioDataSvc->getMetaDataFrame();
-  const auto  collPtr  = evtFrame.get(e4h_coll_name);
+  auto        collPtr  = evtFrame.get(e4h_coll_name);
+  // When the collection can't be retrieved from the frame
+  // there is still the possibility that it was generated
+  // from a functional algorithm
   if (!collPtr) {
-    error() << "No collection with name: " << e4h_coll_name << " available for conversion" << endmsg;
+    DataObject* p;
+    auto        sc  = m_podioDataSvc->retrieveObject(e4h_coll_name, p);
+    auto        ptr = dynamic_cast<AnyDataWrapper<std::shared_ptr<podio::CollectionBase>>*>(p);
+    if (sc.isFailure() || !ptr) {
+      error() << "No collection with name: " << e4h_coll_name << " available for conversion" << endmsg;
+    } else {
+      collPtr = dynamic_cast<podio::CollectionBase*>(ptr->getData().get());
+    }
```

Comment:
Is this path not possible for cases where it is in a Frame? I.e. could we just have one path through this, and drop the usage of the `evtFrame`?

Reply:
I think this question mixes two things: the collection is either in the Frame if it has been produced with the non-functional algorithms or it is not if it has been produced by functional algorithms; then we can not use the event frame by asking the data service for the collections instead (but this could have been done earlier and doesn't have much to do with the current changes) as it is right now.

Reply:
I was more aiming at the question, whether we can reach collections that have been produced by non-functional algorithms via the data svc, instead of going to the event Frame?